src/sentry/utils/not_set.py

```diff
@@ -0,0 +1,10 @@
+from __future__ import annotations
+
+from enum import Enum, auto
+
+
+class NotSet(Enum):
+    TOKEN = auto()
+
+
+NOT_SET = NotSet.TOKEN
```

Comment:
Would be nice to have a docstring somewhere explaining what this is used for if we're tossing it in the kitchen sink